src/NuGet.Clients/VsExtension/VsEvents/OnBuildPackageRestorer.cs

```diff
@@ -819,6 +824,35 @@ private static int GetMSBuildOutputVerbositySetting(DTE dte)
             return 0;
         }
 
+        private IEnumerable<Tuple<string, string>> GetProjectFolderPath()
+        {
+            var projects = _dte.Solution.Projects;
+            foreach (var item in projects)
+            {
+                var project = item as Project;
+
+                if (project != null)
+                {
+                    yield return new Tuple<string, string>(EnvDTEProjectUtility.GetFullPath(project), project.Name);
+                }
+            }
+        }
+
+        private bool CheckPackagesConfig(string folderPath, string projectName)
+        {
+            if (folderPath == null)
+            {
+                return false;
+            }
+            else
+            {
+                return File.Exists(Path.Combine(folderPath, "packages.config"))
+                    || File.Exists(Path.Combine(folderPath, "project.json"))
```

Comment:
you are missing projectname.project.json, and also there should be some sort of a common code to do that, instead of a private method

CC @emgarten